python-packages/smithy-core/smithy_core/types.py

```diff
@@ -41,3 +53,61 @@ def from_json(j: Any) -> "JsonBlob":
         json_string = JsonBlob(json.dumps(j).encode(encoding="utf-8"))
         json_string._json = j
         return json_string
+
+
+class TimestampFormat(Enum):
+    """Smithy-defined timestamp formats with serialization and deserialization helpers.
+
+    See `Smithy's docs <https://smithy.io/2.0/spec/protocol-traits.html#smithy-api-timestampformat-trait>`_
+    for more details.
+    """
+
+    DATE_TIME = "date-time"
+    """RFC3339 section 5.6 datetime with optional millisecond precision but no UTC
+    offset."""
+
+    HTTP_DATE = "http-date"
+    """An HTTP date as defined by the IMF-fixdate production in RFC 7231 section
+    7.1.1.1."""
```

Comment:
Should we be using the latest standard here (RFC 9110 5.6.7) or is there a notable difference in the older RFC we use?

Reply:
I don't think there's any difference, but I can update the reference here and in the smithy docs.

Reply:
https://github.com/smithy-lang/smithy/pull/2298/